com.avaloq.tools.ddk.xtext.format.generator/src/com/avaloq/tools/ddk/xtext/format/generator/FormatFragmentUtil.java

```diff
@@ -64,7 +66,7 @@ private FormatFragmentUtil() {
    */
   private static URI getDefaultFormatLocation(final Grammar grammar, final XpandExecutionContext ctx) {
     final String xmiPath = GrammarUtil.getClasspathRelativePathToXmi(grammar);
-    return URI.createFileURI(new File(ctx.getOutput().getOutlet(Generator.SRC).getPath(), xmiPath).getAbsolutePath()).trimFileExtension().appendFileExtension(FormatConstants.FILE_EXTENSION);

```

Comment:
I think one should change the com.avaloq.tools.ddk.xtext.format.generator.FormatFragment to inherit from org.eclipse.xtext.xtext.generator.AbstractXtextGeneratorFragment instead of the deprecated org.eclipse.xtext.generator.AbstractGeneratorFragment. In the new class one could call getRuntime().getSrc() or similar to access the source folder